### PR TITLE
feat(web.database): remove deprecated calls

### PR DESCRIPTION
**Pull request by mlcdf**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix WEB-15524
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

## Description

Hello,

This PR remove calls from deprecated API and replace them via the new API. That's pretty much it.

This is my first PR on the manager, so let me know if I did anything wrong.

### CODE REVIEW

**Comment by sachinrameshn** [comment]
Hello,

Can you please add ref in your commit message ? 

Example:

feat(universe): description 

ref: ticket#

Sign-off

**Review by darsene — CHANGES_REQUESTED** [review]
Hi @mlcdf ,
The name of your branch is wrong.
If it's new feature, the name has to be feat/MyFeatureRef else fix/MyFeatureRef
For small feature and fix, MyFeatureRef = RefTicket

Other feedback, the description part of the PR has to mentioned which branch is targetted (here `develop`)

File: packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js [hunk 1]
```diff
@@ -115,19 +122,25 @@ angular.module('App').controller(
...
-            this.data.durations.available = duration;
-          },
-        );
+        .getRamPlan(this.productId, this.model.ram.planCode)
+        .then((order) => {
+          this.loading.durations = false;
+
+          const details = order.details.filter((detail) => {
+            return detail.detailType === 'DURATION';
+          });
```
Comment (sachinrameshn) [inline packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js:131]:
```suggestion
          const details = order.details.filter((detail) => {
            return detail.detailType === 'DURATION';
          });
```
const details = order.details.filter(({detailType}) =>
detailType === 'DURATION';
);

Comment (sachinrameshn) [inline packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js:131]:
```suggestion
          const details = order.details.filter(({detailType}) =>
                  detailType === 'DURATION';
);
```

**Review by sachinrameshn — CHANGES_REQUESTED** [review]

File: packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js [hunk 2]
```diff
@@ -40,38 +39,46 @@ angular.module('App').controller(
...
-          );
-          if (this.database.infrastructure === 'legacy') {
-            remove(this.data.availableRam, (ram) => ram === '2048');
-          }
-
-          remove(
-            this.data.availableRam,
-            (ram) => +ram === +this.database.ram.value,
-          );
+          angular.forEach(result, (e) => {
```
Comment (darsene) [inline packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js:63]:
```suggestion
          result.forEach( (e) => {
```

Comment (darsene) [reply packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js:63]: Maybe you should rename as `results`

Comment (sachinrameshn) [inline packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js:63]: Can we retain map here ? result.map()

File: packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js (hunk 1 above)
Comment (mlcdf) [reply packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js:131]: fixed

File: packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js (hunk 2 above)
Comment (mlcdf) [reply packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js:63]: fixed

Comment (mlcdf) [reply packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js:63]: @darsene suggested something else above. Not sure which one of the suggestion I should apply.

**Pull request closed without merging**